### PR TITLE
Alerts v7 New Features Merge

### PR DESCRIPTION
**Pull request by abowersox-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Tests have been added that prove the fix is effective or that the feature works.
- [x] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?

https://jira.carbonblack.local/browse/CBC-30087

## Pull Request Description

Extensions to behavior of Alerts v7.  This is a feature branch merge to develop.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Individual PRs have all been tested and reviewed.

### CODE REVIEW

**Review by kebringer-cb — APPROVED** [review]

File: docs/alerts.rst [hunk 1]
```diff
@@ -153,6 +153,50 @@ You can also read from a csv file by using values that match the profile names i
...
+
+    >>> grouped_alert_facets = group_alert_search_query.facets(["type", "THREAT_ID"], 0, True)
+
+Suppose instead of grouped alerts, you had been working with alerts and wanted to crossover to grouped alerts. Instead of building a new group alert query from scratch you can transform an alert search query into a grouped alert search query or vice versa!
+
+    >>> from cbc_sdk import CBCloudAPI
+    >>> from cbc_sdk.platform import Alert, GroupedAlert
+    >>> api = CBCloudAPI(profile="sample")
+    >>> alert_search_query = api.select(Alert)
+    >>> alert_search_query = grouped_alert_search_query.set_time_range(range="-10d").add_criteria("type", "WATCHLIST").set_minimum_severity(3)
```
Comment (emitreva-cb) [inline docs/alerts.rst:195]: Should this be `grouped_alert_search_query` or just `alert_search_query`?

File: examples/platform/alerts_common_scenarios.py [hunk 2]
```diff
@@ -113,6 +113,7 @@ def main():
     # Device - device - READ: For Device Searches
     # Alerts - org.alerts.close - EXECUTE:
     # Alerts - org.alerts.notes - CREATE, READ, UPDATE, DELETE
+    # Alerts - ThreatMetadata - org.xdr.metadata - READ
```
Comment (emitreva-cb) [inline examples/platform/alerts_common_scenarios.py:116]: This is not part of these changes, but I think for close alert another permission is needed `READ jobs.status`.

File: docs/alerts.rst [hunk 3]
```diff
@@ -153,6 +153,50 @@ You can also read from a csv file by using values that match the profile names i
...
+    >>> group_alert_search_query = alert_search_query.set_group_by()
```
Comment (emitreva-cb) [inline docs/alerts.rst:196]: Can the field arg be omitted?

File: src/cbc_sdk/platform/alerts.py [hunk 4]
```diff
@@ -1510,3 +1632,184 @@ def set_remote_is_private(self, is_private, exclude=False):
...
+        """
+        Closing all alerts matching a grouped alert query is not implemented.
+
+        Note:
+            - Closing all alerts in all groups returned by a ``GroupedAlertSearchQuery`` can be done by
+            getting the ``AlertSearchQuery`` and using close() on it as shown in the following example.
+
+        Example:
+            >>> alert_query = grouped_alert_query.get_alert_search_query()
+            >>> alert_query.close(closure_reason, note, determination)
```
Comment (emitreva-cb) [inline src/cbc_sdk/platform/alerts.py:1762]: The order of the args is different, also do we need the trailing `,` is the parameters list?

File: docs/alerts.rst (hunk 1 above)
Comment (kebringer-cb) [reply docs/alerts.rst:195]: yes, alert_search_query, good pickup.

File: examples/platform/alerts_common_scenarios.py (hunk 2 above)
Comment (kebringer-cb) [reply examples/platform/alerts_common_scenarios.py:116]: yes, thank you

File: docs/alerts.rst (hunk 3 above)
Comment (kebringer-cb) [reply docs/alerts.rst:196]: No, it's needed.

File: src/cbc_sdk/platform/alerts.py (hunk 4 above)
Comment (kebringer-cb) [reply src/cbc_sdk/platform/alerts.py:1762]: fixed both those.

**Review by emitreva-cb — APPROVED** [review]

**Pull request merged**